File: find if path exits in graph.cpp
class Solution {
public:
bool dfs(int u,int tar,vector<int>g[],bool visit[]){
    visit[u]=true;
    if(u==tar){return true;}
    for(int v:g[u]){
        if(!visit[v]){
            if(dfs(v,tar,g,visit)){return true;}
        }
    }
    return false;
}
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {
        vector<int>g[n+1];
        for(vector<int>v:edges){
            int u=v[0],nxt=v[1];
            g[u].emplace_back(nxt);
            g[nxt].emplace_back(u);
        }
        bool visit[n+1];
        memset(visit,false,sizeof(visit));
       queue<int>q;
       q.push(source);
       visit[source]=true;
       while(q.size()){
           int sz=q.size();
           while(sz-->0){
               int cur=q.front();q.pop();
                 if(cur==destination){return true;}
                    for(int v:g[cur]){
                        if(visit[v]==false){
                            q.push(v);
                            visit[v]=true;
                        }
                    }
           }
       }
       return false;
    }
};

File: find if path exits in graph.java
class Solution {
    public boolean validPath(int n, int[][] edges, int source, int destination) {
        ArrayList<ArrayList<Integer>>g=new ArrayList<ArrayList<Integer>>();
        for(int i=0;i<=n;i++){
            g.add(new ArrayList<Integer>());
        }
        int sz=edges.length;
       for(int i=0;i<sz;i++){
           int u=edges[i][0],v=edges[i][1];
           g.get(u).add(v);
           g.get(v).add(u);
       }
       Queue<Integer>q=new LinkedList<Integer>();
       
       q.add(source);
       boolean visit[]=new boolean[n+1];
       visit[source]=true;
       while(q.size()>0){
           int cur=q.peek();q.remove(q.peek());
           if(cur==destination){return true;}
           for(int nxt:g.get(cur)){
               if(visit[nxt]==true){continue;}
               q.add(nxt);
               visit[nxt]=true;
           }
       }
       return false;
    }
}